### PR TITLE
Added lesson 3 exercises 1-3 for review.

### PR DESCRIPTION
**Pull request by Brent-K**

### CODE REVIEW

File: students/brent_kieszling/Lesson_03/Assignment_01/mailroom_part_01.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            
+        #Display report
+        elif choice == 2:
+            #Retrive donor stats
+            donor_info = donor_stats(lst_donors)
+            display_report(donor_info)
+            
+        else:
+            print("That is not a valid option.")
+
```
Comment (charlesrobison) [inline students/brent_kieszling/Lesson_03/Assignment_01/mailroom_part_01.py:188]: Looks good and works great!

File: students/brent_kieszling/Lesson_03/Exercise_3_1/slicing_lab.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+assert remove_ends_4(a_tuple) == ()
+assert remove_ends_4(a_long_tuple) == (5, 6, 7, 8)
+
+assert reverse(a_string) == "gnirts a si siht"
+assert reverse(a_tuple) == (32, 5, 12, 13, 54, 2)
+
+assert swap_thirds(a_string) == "tringthis is a s"
+assert swap_thirds(a_tuple) == (5, 32, 2, 54, 13, 12)
+
+print("Tests passed.")
```
Comment (charlesrobison) [inline students/brent_kieszling/Lesson_03/Exercise_3_1/slicing_lab.py:107]: Nice!

File: students/brent_kieszling/Lesson_03/Exercise_3_2/list_lab.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    
+    Args:
+        lst_fruits (list): list to be manipulated.
+    """
+    mod_fruits = lst_fruits[:]
+    print(mod_fruits)
+    new_fruit = input("What fruit would you like to add? ")
+    mod_fruits.append(new_fruit)
+    print("The new list is: \n {}".format(mod_fruits))
+    select_fruit = int(input("Please select a fruit to inspect (1-{}) ".format(len(mod_fruits))))
```
Comment (charlesrobison) [inline students/brent_kieszling/Lesson_03/Exercise_3_2/list_lab.py:26]: Good

File: students/brent_kieszling/Lesson_03/Exercise_3_2/list_lab.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    print()
+    
+    print("Adding Melon to the front of the list")
+    mod_fruits.insert(0, "Melon")
+    print(mod_fruits)
+    print()
+    
+    print("Displaying all elements that start with P")
+    for item in mod_fruits:
+        if item[0] == "P":
```
Comment (charlesrobison) [inline students/brent_kieszling/Lesson_03/Exercise_3_2/list_lab.py:43]: Recommend upper() or the like to catch case sensitive user input

File: students/brent_kieszling/Lesson_03/Exercise_3_2/list_lab.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+            a = input("Do you like {}? ".format(item.lower()).lower())
+            if a == "yes":
+                break
+            elif a == "no":
+                print("Removing {}.".format(item.lower()))
+                mod_fruits.remove(item)
+                break
+            else:
+                print("Please use 'yes' or 'no'.")
+    print(mod_fruits)
```
Comment (charlesrobison) [inline students/brent_kieszling/Lesson_03/Exercise_3_2/list_lab.py:89]: Note this is skipping an item on the list when user input is 'no' - can you see why?

**Review by charlesrobison — APPROVED** [review]
Great work here!  Just a few comments, nothing major.

Ready to merge

**Comment by Brent-K** [comment]
For exercise 3-2 I think I am skipping an item because when the item is removed the list shrinks. So when 'no' is input for Apples, it is removed and Pears now lives at mod_fruits[0] then we move to the next position  mod_fruits[1] which is now occupied by Oranges. This means Pears never gets evaluated. Instead, would below be a better solution?> +        lst_fruits (list): list to be manipulated.
+    """
+    mod_fruits = lst_fruits[:]
+    print(mod_fruits)
+    for item in mod_fruits:
+        while True:
+            a = input("Do you like {}? ".format(item.lower()).lower())
+            if a == "yes":
+                break
+            elif a == "no":
+                print("Removing {}.".format(item.lower()))
+                lst_fruits.remove(item)
+                break
+            else:
+                print("Please use 'yes' or 'no'.")
+    print(lst_fruits)This way I would move through the copy without being impacted by the remove call to the original list.
   On Thursday, November 12, 2020, 7:24:52 PM PST, Charles Robison <notifications@github.com> wrote:  
 
 


@charlesrobison approved this pull request.

Great work here! Just a few comments, nothing major.

Ready to merge

In students/brent_kieszling/Lesson_03/Assignment_01/mailroom_part_01.py:
> +                    #Retrieve individual's donation history
+                    donor = find_donor(lst_donors, name)
+                    #Display thank you
+                    print(thank_you(donor))
+                    print()
+                    break
+            
+        #Display report
+        elif choice == 2:
+            #Retrive donor stats
+            donor_info = donor_stats(lst_donors)
+            display_report(donor_info)
+            
+        else:
+            print("That is not a valid option.")
+

Looks good and works great!

In students/brent_kieszling/Lesson_03/Exercise_3_1/slicing_lab.py:
> +assert exchange_first_last(a_tuple) == (32, 54, 13, 12, 5, 2)
+
+assert remove_every_other(a_string) == "ti sasrn"
+assert remove_every_other(a_tuple) == (2, 13, 5)
+
+assert remove_ends_4(a_string) == " is a st"
+assert remove_ends_4(a_tuple) == ()
+assert remove_ends_4(a_long_tuple) == (5, 6, 7, 8)
+
+assert reverse(a_string) == "gnirts a si siht"
+assert reverse(a_tuple) == (32, 5, 12, 13, 54, 2)
+
+assert swap_thirds(a_string) == "tringthis is a s"
+assert swap_thirds(a_tuple) == (5, 32, 2, 54, 13, 12)
+
+print("Tests passed.")

Nice!

In students/brent_kieszling/Lesson_03/Exercise_3_2/list_lab.py:
> +fruits = ["Apples", "Pears", "Oranges", "Peaches"]
+
+
+#PROCESS------------------------------------
+def series_1(lst_fruits):
+    """Perform a series of operations on a list.
+    
+    Args:
+        lst_fruits (list): list to be manipulated.
+    """
+    mod_fruits = lst_fruits[:]
+    print(mod_fruits)
+    new_fruit = input("What fruit would you like to add? ")
+    mod_fruits.append(new_fruit)
+    print("The new list is: \n {}".format(mod_fruits))
+    select_fruit = int(input("Please select a fruit to inspect (1-{}) ".format(len(mod_fruits))))

Good

In students/brent_kieszling/Lesson_03/Exercise_3_2/list_lab.py:
> +    print()
+    
+    print("Adding Kiwi to the front of the list")
+    a = ["Kiwi"]
+    mod_fruits = a + mod_fruits
+    print(mod_fruits)
+    print()
+    
+    print("Adding Melon to the front of the list")
+    mod_fruits.insert(0, "Melon")
+    print(mod_fruits)
+    print()
+    
+    print("Displaying all elements that start with P")
+    for item in mod_fruits:
+        if item[0] == "P":

Recommend upper() or the like to catch case sensitive user input

In students/brent_kieszling/Lesson_03/Exercise_3_2/list_lab.py:
> +        lst_fruits (list): list to be manipulated.
+    """
+    mod_fruits = lst_fruits[:]
+    print(mod_fruits)
+    for item in mod_fruits:
+        while True:
+            a = input("Do you like {}? ".format(item.lower()).lower())
+            if a == "yes":
+                break
+            elif a == "no":
+                print("Removing {}.".format(item.lower()))
+                mod_fruits.remove(item)
+                break
+            else:
+                print("Please use 'yes' or 'no'.")
+    print(mod_fruits)

Note this is skipping an item on the list when user input is 'no' - can you see why?

**Pull request merged**